addons/hr/models/res_users.py

```diff
@@ -361,3 +362,21 @@ def action_related_contact(self):
             'res_model': 'res.partner',
             'view_mode': 'form',
         }
+
+    def _get_store_avatar_card_fields(self):
+        avatar_card_fields = super()._get_store_avatar_card_fields()
+        # HACK: fetch the employee fields from employees to retrieve hr.employee.public fields if no access to hr.employee
+        self.employee_ids.fetch(
+            [
+                field.field_name if isinstance(field, Store.Attr) else field
+                for field in self.env[
+                    "hr.employee"
+                ]._get_store_avatar_card_fields()
```

Comment:
_get_store_avatar_card_fields of employee could be called a single time in this function and store the value in a variable